runtime/src/secondary_index.rs

```diff
@@ -99,12 +103,125 @@ impl SecondaryIndexEntry for RwLockSecondaryIndexEntry {
     }
 }
 
+#[derive(Debug, Default)]
+struct HierarchicalOrderedMap<L1, L2>
+where
+    L1: Default + Sync + Send + Copy + PartialEq + Ord,
+    L2: Default + Sync + Send + Copy + PartialEq + Ord,
```

Comment:
Can we avoid requiring Copy here?

Reply:
IDK can we? Like how do you do that?

Reply:
I would remove `Copy` and see what if anything the compiler complains about. How did you generate this list of trait requirements?

Reply:
OHHH the copy trait! My bad thought you meant not to specify for both K, V. Like do some sorta `K, V: (list of traits)` C style thing.

Reply:
Breaks pretty badly when I get rid of it. Why remove it? Any particular reason?

Reply:
Yes, we may remove the derive of `Copy` from Pubkey, at which point this struct will become unusable for your index.
You may need to go down the rabbit hole a little, removing `Copy` from various places and potentially fixing up some of the code.

Reply:
Looks lie it could be removed if your comment about the sub-slice worked, but its needed for that code section as there is the temp copy that it needs to function.